### PR TITLE
refactored lib_wahoo_install

### PR DESCRIPTION
**Pull request by gretel**
use depth of 1 for all 'git clone' operations
refactored `lib_wahoo_install`:
* removed 'forking' part
* smoothly set remote 'upstream'
* save some LOC (`sh` has POSIX-like `&&` :cat:)
* lower logging level for configuration writes but keep a warning on writing back-up copy
* removed `INIT` (only used once)
* introduced `FISH_CONFIG_FILE`
* unified logging style
* minor move-arounds

### CODE REVIEW

**Comment by bucaran** [comment]
@gretel Thanks! Looks good to me. Did you test this on a new install?

File: bootstrap.sh [hunk 1]
```diff
@@ -157,7 +157,7 @@ lib_fish_install() {
     fi
 
     mkdir -p "${HOME}/src"
-    git clone $FISH_REMOTE "${HOME}/${FISH_SOURCE}"
+    git clone -q --depth 1 $FISH_REMOTE "${HOME}/${FISH_SOURCE}"
```
Comment (bucaran) [inline bootstrap.sh]: Why `--depth 1`?

File: bootstrap.sh [hunk 2]
```diff
@@ -250,103 +250,67 @@ lib_autoconf_install() {
...
-    BASE="${HOME}"
-  fi
-
+  test -z ${BASE+_} && BASE="${HOME}"
   util_log INFO "Resolving Wahoo path → ${BASE}/.wahoo"
 
   if [ -d "${BASE}/.wahoo" ]; then
     util_log ERROR "Wahoo is already installed."
-    util_log INFO "If you want to update Wahoo try: 'fish update'"
+    util_log INFO "If you would like to update Wahoo please call \"fish -c 'wa update\"."
```
Comment (bucaran) [inline bootstrap.sh]: We don't need to be so polite at this level don't you think?

> If you would like to update Wahoo run `wa update`"

What is this `fish -c 'wa update'`? :-1:

File: bootstrap.sh [hunk 3]
```diff
@@ -250,103 +250,67 @@ lib_autoconf_install() {
...
-    # curl -u "$GITHUB_USER" --fail --silent \
-    #   https://api.github.com/repos/wahoo/forks \
-    #   -d "{\"user\":\"$GITHUB_USER\"}" >/dev/null ^&1
+  pushd ${BASE}/.wahoo >/dev/null 2>&1
+  REF=$(git config remote.upstream.url)
+  if [ -z "${REF}" ]; then
+    git remote add upstream $URL
   else
-    util_log WARN "GitHub user information not detected."
+    git remote set-url upstream $URL
```
Comment (bucaran) [inline bootstrap.sh]: Is it possible to just run `git remote set-url upstream $URL` or will that fail is there is no remote registered already?

File: bootstrap.sh [hunk 4]
```diff
@@ -250,103 +250,67 @@ lib_autoconf_install() {
...
-
+  | sed -e "s|$HOME|\$HOME|")" > ${FISH_CONFIG_FILE}
   echo "set -g WAHOO_CUSTOM $(echo "${CUSTOM}" \
-  | sed -e "s|$HOME|\$HOME|")" >> "${FISH_CONFIG}/config.fish"
-
-  echo "source \$WAHOO_PATH/${INIT}" >> "${FISH_CONFIG}/config.fish"
-
-  WAHOO_CONFIG="${HOME}/.config/wahoo"
+  | sed -e "s|$HOME|\$HOME|")" >> ${FISH_CONFIG_FILE}
+  echo "source \$WAHOO_PATH/init.fish" >> ${FISH_CONFIG_FILE}
```
Comment (bucaran) [inline bootstrap.sh]: Have you tested that everything still works with these minor changes?

File: bootstrap.sh (hunk 1 above)
Comment (gretel) [inline bootstrap.sh]: let me answer with a question - why clone with history as long as n?

File: bootstrap.sh (hunk 2 above)
Comment (gretel) [inline bootstrap.sh]: another question: what is with `fish update` - i guess that never did work?

File: bootstrap.sh (hunk 3 above)
Comment (gretel) [inline bootstrap.sh]: why should i have the `if` block then? why does `git remote` have different commands, if one would be the same as another?

File: bootstrap.sh (hunk 4 above)
Comment (gretel) [inline bootstrap.sh]: did you notice the according `travis` run?

**Comment by gretel** [comment]
https://github.com/fish-shell/wahoo/issues/63

File: bootstrap.sh (hunk 1 above)
Comment (bucaran) [inline bootstrap.sh]: :+1:

File: bootstrap.sh (hunk 2 above)
Comment (bucaran) [inline bootstrap.sh]: It's not about politeness, but making messages less verbose. 

`If you would like to update Wahoo please run wa update"` :+1: 

I still don't understand what `fish update` should do.

File: bootstrap.sh (hunk 3 above)
Comment (bucaran) [inline bootstrap.sh]: So, I guess the answer is not possible. :+1:

File: bootstrap.sh (hunk 4 above)
Comment (bucaran) [inline bootstrap.sh]: :+1:

File: bootstrap.sh (hunk 2 above)
Comment (gretel) [inline bootstrap.sh]: please explain to me what `fish update` is about

Comment (bucaran) [inline bootstrap.sh]: Wahoo should update with `wa update`.

Comment (bucaran) [inline bootstrap.sh]:
```fish
 if [ -d "${BASE}/.wahoo" ]; then
     util_log ERROR "Wahoo is already installed."
     util_log INFO "Run `wa update` to update WAhoo."
```

Comment (gretel) [inline bootstrap.sh]: well the call does `wa update`, doesn't it? i don't get your last quote in regards to that

**Comment by bucaran** [comment]
#63 Done.

Please address the 

```fish
f [ -d "${BASE}/.wahoo" ]; then
     util_log ERROR "Wahoo is already installed."
     util_log INFO "Run `wa update` to update Wahoo."
...
```

and let's merge this.

**Comment by gretel** [comment]
@bucaran added https://github.com/gretel/wahoo/commit/d8a03f954043b2644c678a9b10c8e969e4dfc9a7

**Comment by gretel** [comment]
so do we hit the green button or not?

**Comment by bucaran** [comment]
:+1:

**Pull request merged**